server/src/main/java/io/helidon/extensions/mcp/server/McpSampling.java

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+package io.helidon.extensions.mcp.server;
+
+import java.lang.System.Logger.Level;
+import java.util.function.Consumer;
+
+import io.helidon.http.sse.SseEvent;
+import io.helidon.webserver.sse.SseSink;
+
+import jakarta.json.JsonObject;
+
+import static io.helidon.extensions.mcp.server.McpJsonRpc.createSamplingRequest;
+import static io.helidon.extensions.mcp.server.McpJsonRpc.createSamplingResponse;
+import static io.helidon.extensions.mcp.server.McpJsonRpc.prettyPrint;
+
+/**
+ * MCP Sampling feature.
+ */
+public final class McpSampling extends McpFeature {
+    private static final System.Logger LOGGER = System.getLogger(McpSampling.class.getName());
+
+    McpSampling(McpSession session) {
+        super(session);
+    }
+
+    McpSampling(McpSession session, SseSink sseSink) {
+        super(session, sseSink);
+    }
+
+    /**
+     * Whether the connected client supports sampling feature.
+     *
+     * @return {@code true} if the connected client supports sampling feature,
+     * {@code false} otherwise.
+     */
+    public boolean isEnabled() {
+        return session()
+                .capabilities()
+                .contains(McpCapability.SAMPLING);
+    }
+
+    /**
+     * Send the provided sampling request to the client and return its response.
+     *
+     * @param request sampling request
+     * @return sampling response
+     * @throws io.helidon.extensions.mcp.server.McpSamplingException when an error occurs
+     */
+    public McpSamplingResponse request(Consumer<McpSamplingRequest.Builder> request) throws McpSamplingException {
+        var builder = McpSamplingRequest.builder();
+        request.accept(builder);
+        return request(builder.build());
+    }
+
+    /**
+     * Send the provided sampling request to the client and return its response.
+     *
+     * @param request sampling request
+     * @return sampling response
+     * @throws io.helidon.extensions.mcp.server.McpSamplingException when an error occurs
+     */
+    public McpSamplingResponse request(McpSamplingRequest request) throws McpSamplingException {
+        long id = session().jsonRpcId();
+        JsonObject payload = createSamplingRequest(id, request);
+
+        log("Sampling request:\n" + prettyPrint(payload));
+        sseSink().ifPresentOrElse(sink -> sink.emit(SseEvent.builder()
+                                            .name("message")
+                                            .data(payload)
+                                            .build()),
+                                  () -> session().send(payload));
+        JsonObject response = session().pollResponse(id, request.timeout());
+        log("Sampling response:\n" + prettyPrint(response));
+        return createSamplingResponse(response);
+    }
+
+    private void log(String message) {
+        if (LOGGER.isLoggable(Level.DEBUG)) {
+            LOGGER.log(Level.DEBUG, () -> message);
```

Comment:
This method is not great. The reason you guard by `isLoggable` is not to resolve string concatenation etc.

For example on line 87, you will do the `prettyPrint(response)` for every single request, even if debug is not enabled.

Please remove this method, and guard each log statement.
Also if you guard the log statement by `isLoggable`, do not use the lambda version of the method